src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -45,27 +45,26 @@ def __init__(self, cloud_sync):
 
         self.provider = REMOTES[self.cloud_sync["credentials"]["provider"]]
 
+        self.config = None
         self.tmp_file = None
         self.tmp_file_exclude = None
 
-        self.path = None
-
-    def __enter__(self):
+    async def __aenter__(self):
         self.tmp_file = tempfile.NamedTemporaryFile(mode="w+")
 
         # Make sure only root can read it as there is sensitive data
         os.chmod(self.tmp_file.name, 0o600)
 
         config = dict(self.cloud_sync["credentials"]["attributes"], type=self.provider.rclone_type)
-        config = dict(config, **self.provider.get_credentials_extra(self.cloud_sync["credentials"]))
+        config = dict(config, **await self.provider.get_credentials_extra(self.cloud_sync["credentials"]))
         if "pass" in config:
             config["pass"] = rclone_encrypt_password(config["pass"])
```

Comment:
Are other methods in here safe to run in event loop? e.g. rclone_encrypt_password

Reply:
`rclone_encrypt_password` yes, writing temporary files in ramfs - subject to discuss